htscodecs/c_range_coder.h

```diff
@@ -61,13 +67,39 @@ static inline void RC_StartDecode(RangeCoder *rc)
     rc->Carry = 0;
     rc->Cache = 0;
     rc->code  = 0;
+    rc->err   = 0;
     if (rc->in_buf+5 > rc->in_end) {
         rc->in_buf = rc->in_end; // prevent decode
         return;
     }
     DO(5) rc->code = (rc->code<<8) | *rc->in_buf++;
 }
 
+static inline void RC_ShiftLowCheck(RangeCoder *rc) {
+    if (rc->low < Thres || rc->Carry) {
+        if (rc->out_end && rc->out_buf + rc->FFNum >= rc->out_end) {
```

Comment:
Strictly, to avoid UB this should be:
```suggestion
        if (rc->out_end && rc->FFNum >= rc->out_end - rc->out_buf) {
```